lib/uploader/uploader.ts

```diff
@@ -491,8 +492,16 @@ export class Uploader {
 			if (!disabledChunkUpload) {
 				logger.debug('Initializing chunked upload', { file, upload })
 
+				let chunksWorkspace: string
+				if (this._isPublic) {
+					const shareToken = root!.split('/').at(-1)!
+					chunksWorkspace = `${getBaseUrl()}/public.php/dav/uploads/${shareToken}`
+				} else {
+					chunksWorkspace = generateRemoteUrl(`dav/uploads/${getCurrentUser()?.uid}`)
+				}
+
```

Comment:
I do not belive this belongs to the uploader. Rather this is exactly what `initChunkWorkspace` should care about.